packages/passport/sdk-sample-app/src/components/LinkWallet.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useState } from 'react';
+import {
+  Button, Form, Offcanvas, Spinner, Stack,
+} from 'react-bootstrap';
+import { ModalProps } from '@/types';
+import { Heading } from '@biom3/react';
+import { generateNonce } from 'siwe';
+
+function LinkWallet({ showModal, setShowModal, onSubmit }: ModalProps & {
+  onSubmit: (type: string, walletAddress: string, signature: string, nonce: string) => void;
+}) {
+  const [type, setType] = useState<string>('');
+  const [walletAddress, setWalletAddress] = useState<string>('');
+  const [signature, setSignature] = useState<string>('');
+  const [isSubmitting, setIsSubmitting] = useState<boolean>(false);
+  const [isInvalid, setInvalid] = useState<boolean | undefined>(undefined);
+
+  const resetForm = () => {
+    setType('');
+    setWalletAddress('');
+    setSignature('');
+  };
+
+  const handleClose = () => {
+    resetForm();
+    setInvalid(false);
+    setIsSubmitting(false);
+    setShowModal(false);
+  };
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    e.stopPropagation();
+
+    const form = e.currentTarget;
+    if (form.checkValidity()) {
+      setInvalid(false);
+      setIsSubmitting(true);
+      try {
+        const nonce = generateNonce();
```

Comment:
The nonce will need to be provided as input as well. It's in the typed data that gets signed so it will be in the signature.